src/lamp_py/ingestion/glides.py

```diff
@@ -15,41 +16,67 @@
 from lamp_py.aws.s3 import download_file, upload_file
 from lamp_py.aws.kinesis import KinesisReader
 from lamp_py.ingestion.utils import explode_table_column, flatten_table_schema
+from lamp_py.utils.dataframely import unnest_columns
 from lamp_py.runtime_utils.process_logger import ProcessLogger
 from lamp_py.runtime_utils.remote_files import (
     LAMP,
     S3_SPRINGBOARD,
 )
 
+RFC3339_TIME_REGEX = r"[0-9]T[012][0-9]:[0-5][0-9]:[0-6][0-9](.[0-9]*)?(Z|[+-][012][0-9]:[0-5][0-9])$"
+GTFS_TIME_REGEX = r"^([0-9]{2}):([0-5][0-9]):([0-5][0-9])$"  # clock can be greater than 24 hours
+RFC3339_DATE_REGEX = r"^[0-9]{4}-[01][0-9]-[0-3]"
+
 
 class GlidesConverter(ABC):
     """
     Abstract Base Class for Archiving Glides Events
     """
 
-    glides_user = pyarrow.struct(
-        [
-            ("emailAddress", pyarrow.string()),
-            ("badgeNumber", pyarrow.string()),
-        ]
+    user = dy.Struct(
+        {
+            "emailAddress": dy.String(metadata={"reader_roles": ["GlidesUserEmail"]}),
+            "badgeNumber": dy.String(),
+        }
     )
 
-    glides_location = pyarrow.struct(
-        [
-            ("gtfsId", pyarrow.string()),
-            ("todsId", pyarrow.string()),
-        ]
+    location = dy.Struct({"gtfsId": dy.String(nullable=True), "todsId": dy.String(nullable=True)})
+
+    metadata = dy.Struct(
+        {
+            "location": location,
+            "author": user,
+            "inputType": dy.String(),
+            "inputTimestamp": dy.String(regex=RFC3339_DATE_REGEX + RFC3339_TIME_REGEX),
+        }
     )
 
-    glides_metadata = pyarrow.struct(
-        [
-            ("location", glides_location),
-            ("author", glides_user),
-            ("inputType", pyarrow.string()),
-            ("inputTimestamp", pyarrow.string()),
-        ]
+    trip_key = dy.Struct(
+        {
+            "serviceDate": dy.String(nullable=True),
+            "tripId": dy.String(nullable=True),
+            "startLocation": location,
+            "endLocation": location,
+            "startTime": dy.String(nullable=True, regex=GTFS_TIME_REGEX),
+            "endTime": dy.String(nullable=True, regex=GTFS_TIME_REGEX),
+            "revenue": dy.String(nullable=True, regex=r"(non)?revenue"),
+            "glidesId": dy.String(),
+        }
     )
 
+    class Record(dy.Schema):
```

Comment:
Can we give this class a more descriptive name than `Record`? It's glides specific, and also pretty raw, or unprocessed. 


```suggestion
    class GlidesSourceRecord(dy.Schema):
```